stock_location_package_restriction/models/stock_move.py

```diff
@@ -1,112 +1,21 @@
-                location = new_location
-            existing_package_ids = location_packages.get(location.id, set())
-            existing_packages = Package.browse(existing_package_ids).exists()
-            new_packages = Package.browse()
-            for move_line in move_lines:
-                package = move_line.result_package_id
-                # CASE: Package restiction exists but there is no package in move
-                if not package:
-                    error_msgs.append(
-                        _(
-                            "A package is mandatory on the location {location}. "
-                            "You cannot move the product {product} without a package."
-                        ).format(
-                            location=location.display_name,
-                            product=move_line.product_id.display_name,
-                        )
-                    )
-                    continue
-                if location.package_restriction != SINGLEPACKAGE:
-                    continue
-                # CASE: Package on location, new package is different
-                if existing_package_ids and package.id not in existing_package_ids:
-                    error_msgs.append(
-                        _(
-                            "Only one package is allowed on the location {location}."
-                            "You cannot add the {package}, there is already "
-                            "{existing_packages}."
-                        ).format(
-                            location=location.display_name,
-                            existing_packages=", ".join(
-                                existing_packages.mapped("name")
-                            ),
-                            package=package.display_name,
-                        )
-                    )
-                    continue
-                # CASE: Multiple packages in Move but single location
-                # with singlepackage Restriction
-                if new_packages and package not in new_packages:
-                    error_msgs.append(
-                        _(
-                            "Only one package is allowed on the location {location}."
-                            "You cannot move multiple packages to it:"
-                            "{packages}"
-                        ).format(
-                            location=location.display_name,
-                            packages=", ".join(new_packages.mapped("display_name")),
-                        )
-                    )
-                    continue
-                new_packages |= package
-
-        if error_msgs:
-            raise ValidationError("\n".join(error_msgs))
+    def _check_location_package_restriction(self):
+        """Check if the moves can be done regarding potential package restrictions"""
+        self.filtered(
+            lambda m: m.state == "done"
+        ).location_dest_id._check_package_restriction()
```

Comment:
This is now always checking all quants on the location. Should it not only check the current move lines?
```suggestion
        ).location_dest_id._check_package_restriction(self.move_line_ids)
```

Reply:
If the location has a violation but the just moved move lines are correct then you would accept to put them on the location. I thought it is safer to check the entire location.

Reply:
Yes it would be safer, but this would change the current implementation. 
At the moment it is to book a moves on a location, although the location has a violation.